fe1-web/src/core/objects/Hash.ts

```diff
@@ -78,4 +80,37 @@ export class Hash extends Base64UrlData {
     const byteArray = new TextEncoder().encode(str);
     return byteArray.length;
   }
+
+  /**
+   * Returns *some* string representation of this object.
+   * If you need access to the unterlying data type use .valueOf() and
+   * if you want to serialize an instance use .toState() instead
+   */
+  public toString(): string {
+    return super.toString();
+  }
+
+  /**
+   * Returns the primitive value used for representing the Hash,
+   * a string
+   * If you want to serialize an instance use .toState() instead
+   */
+  public valueOf(): string {
+    return super.valueOf();
+  }
+
+  /**
+   * Returns the serialized version of the hash that can for instance be stored
+   * in redux stores
+   */
+  public toState(): HashState {
```

Comment:
I do agree that it gives flexibility, but I don't see the state fundamentally changing, given the `Hash` itself is a fancy encapsulation of a string. 
It seems that `string` here would make it inheritable (one to one) from Base64UrlData